src/main.tf

```diff
@@ -27,25 +17,30 @@ resource "azurerm_resource_group" "main" {
 }
 
 resource "azurerm_cosmosdb_account" "main" {
-  name                              = var.md_metadata.name_prefix
-  location                          = azurerm_resource_group.main.location
-  resource_group_name               = azurerm_resource_group.main.name
-  offer_type                        = local.offer_type
-  kind                              = local.kind
-  is_virtual_network_filter_enabled = true
+  name                               = var.md_metadata.name_prefix
+  location                           = azurerm_resource_group.main.location
+  resource_group_name                = azurerm_resource_group.main.name
+  offer_type                         = "Standard"
+  kind                               = "MongoDB"
+  is_virtual_network_filter_enabled  = true
+  public_network_access_enabled      = false
+  access_key_metadata_writes_enabled = false
+  key_vault_key_id                   = azurerm_key_vault_key.main.resource_versionless_id
 
   enable_automatic_failover       = var.geo_redundancy.automatic_failover
   enable_multiple_write_locations = var.geo_redundancy.multi_region_writes
   mongo_server_version            = var.database.mongo_server_version
 
-  virtual_network_rule {
-    id = azurerm_subnet.main.id
-  }
-
+  # If we wanted to use RBAC instead of access policies for CMK, we would need to set up a two-step deployment for:
```

Comment:
is this a TODO or just a lesson you learned while authoring this PR? 
if the former, let's make a GH issue.
if the later, I don't think this needs to be part of the codebase as a reader is just prone to think "idk should we use RBAC instead?".